README.md

```diff
@@ -2,28 +2,103 @@
-embedded software development cycle.
-
-Using Flipper, a developer creates and debugs program logic on a development
-machine instead of deploying code onto an embedded device and debugging it
-there. Our library performs remote procedure calls to a connected device
-instead of emulating the hardware. This makes it possible to use real hardware
-to test program behavior in real time. When development is complete, the
-project can be cross compiled and loaded onto a Flipper device for native
-performance.
-
-This new embedded development workflow makes it simple to use widely adopted
-and industry standard tools, like Xcode and Visual Studio, to develop and debug
-applications that interact with embedded hardware peripherals. The capabilities
-of the platform extend beyond the scope of embedded software development;
-Flipper makes it easy to write applications in any programming language, on any
-platform, that control real hardware.
-
-To purchase a Flipper device, please visit our store
-[here](https://flipper.io/products/flipper-carbon-developer-unit).
+An embedded development platform that can be controlled remotely from any
+programming language. [flipper.io](https://flipper.io)
+
+# Why did we make Flipper?
+
+At the highest level, Flipper is just like any embedded development platform:
+it's a tool for using software to interact with and control electronic
+hardware. All embedded platforms have limited resources, often orders of
+magnitude smaller than those available on a typical desktop computer. These
+limitations impose heavy constraints on the type of applications that you can
+write for these platforms. For example, nearly all embedded programs are
+written in C or C++
+([though Rust is getting there](http://www.rust-embedded.org/)), many lack
+access to resources like heap memory, and simple tasks like communicating with
+other devices can involve a lot of manual protocol implementation. Quite
+frankly, embedded development is really hard, especially for beginners. With
+Flipper, we're trying to make embedded development easier by moving the
+application logic to a more comfortable place - the developer's computer.
+
+# What's special about Flipper?
+
+Flipper has two core components: a lightweight operating system for the
+device, and a library to install on "host machines", such as the developer's
+computer. Together, these form a system that we like to call the Flipper
+Message Runtime, or FMR. FMR allows you to write programs that execute on the
+host machine but directly control the hardware. I like to think of it as
+hardware-as-a-library. This ability to remotely execute functions on the
+device from a host machine has far-reaching implications, and opens up a
+whole new world of possibilities for embedded projects. The most exciting
+feature of Flipper is the ability to control it from any platform (Linux,
+MacOS, Windows, Android, or iOS), using any programming language.
+
+## An example
```

Comment:
I think this would be better deferred to a readme in the examples folder.

Reply:
I think that people like to see how things work, and that examples and specific use cases draw them in. My favorite repositories to read through are the ones centered around "here's what YOU as the consumer of this code would do". I think that code examples in the top readme are not only a good idea, but very necessary to maintain people's attention and interest. You and I know what to expect when we talk about packages or language bindings, but nobody else will know what to picture in their minds. I don't even think this particular example is 100% accurate, but it's useful because it helps to build the understanding of how users will write code and how the code interacts in the overall system.